dbcon/mysql/ha_calpont_execplan.cpp

```diff
@@ -5150,6 +5190,16 @@ void gp_walk(const Item* item, void* arg)
 
                 case STRING_RESULT:
                 {
+                    // Special handling for 0xHHHH literals
+                    const Type_handler *tph = item->type_handler();
+                    if (dynamic_cast<const Type_handler_hex_hybrid*>(tph) != NULL)
```

Comment:
That is cheaper to use typeid(*tph)==typeid(Type_handler_hex_hybrid) b/c dynamic_cast recurses through the class hierarchy. However tph could be of a different class that inherits from Type_handler_hex_hybrid. If the suggestion is applicable then use it.

Reply:
This is a good idea. Done.